exampleCourse/questions/templateRandomMultipleChoiceWithFigure/server.py

```diff
@@ -0,0 +1,22 @@
+import random
+
+def generate(data):
+
+    #Create a list with the file name of all UIUC figures
+    allUIUCfigures = ["fig"+str(i)+".jpeg" for i in range(1,8)]
+
+    #Make a random selection of 3 figures from UIUC
+    trueFig = random.sample(allUIUCfigures,3)
+
+    #Store the true figures in data['params']
+    for i,fig in enumerate(trueFig):
+        data['params']['filename'+str(i+1)] = fig
```

Comment:
```suggestion
    #Create a list with the file name of all UIUC figures
    UIUCfigures = ["fig1.jpeg", "fig2.jpeg", "fig3.jpeg", "fig4.jpeg", "fig5.jpeg", "fig6.jpeg", "fig7.jpeg", "fig8.jpeg"]

    # Shuffle the order of the UIUC figures
    random.shuffle(UIUCfigures)

    # Store the first three shuffled UIUC figures as the wrong answers
    data['params']['wrong_filename0'] = UIUCfigures[0]
    data['params']['wrong_filename1'] = UIUCfigures[1]
    data['params']['wrong_filename2'] = UIUCfigures[2]
```

I'm thinking that something like this ^^^ might be easier for non-Python-experts to follow?

Reply:
haha. This is actually funny! I never thought I would one day get that type of advice (the non_Python expert here speaking). Yes, sure, we can change that.